redaxo/src/addons/phpmailer/lib/cronjob.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * PHPMailer Addon.
+ *
+ * @author markus[dot]staab[at]redaxo[dot]de Markus Staab
+ *
+ * @package redaxo\phpmailer
+ */
+
+class rex_cronjob_mailer_purge extends rex_cronjob
+{
+    /**
+     * purgeMailarchive.
+     *
+     * @return int
+     */
+    protected function purgeMailarchive(int $days = 7, string $dir = '', int $log = 0)
+    {
+        $files = glob($dir . '/*');
+        $file = '';
+        if ($files) {
+            foreach ($files as $file) {
+                if (is_dir($file)) {
+                    $log = $log + self::purgeMailarchive($days, $file);
+                } elseif ((time() - filemtime($file)) > (60 * 60 * 24 * $days)) {
+                    if (rex_file::delete($file)) {
+                        ++$log;
+                    }
+                }
+            }
+            if ('' != $dir && $dir != rex_mailer::logFolder() && is_dir($dir)) {
+                if (0 === count(glob("$dir/*")) && true == rmdir($dir)) {
+                    ++$log;
+                }
+            }
+        }
+        return $log;
+    }
+
+    public function execute()
+    {
+        $logfolder = rex_mailer::logFolder();
+        if ('' != $logfolder && is_dir($logfolder)) {
+            $days = (int) $this->getParam('days');
+            $purgeLog = self::purgeMailarchive($days, $logfolder);
+            if (0 != $purgeLog) {
+                $this->setMessage('Objekte gelöscht: '.$purgeLog);
+                return true;
+            }
+            $this->setMessage(rex_i18n::msg('phpmailer_archivecron_nothing_to_delete'));
```

Comment:
Cronjob-Logmeldungen machen wir bisher in englisch, ohne Übersetzung. Die Sprache ist ja abhängig vom gerade eingeloggten Benutzer, während der Job ausgeführt wird. Die Sprache könnte somit im Cronjob-Log immer mal wechseln, wenn wir es mit Übersetzung machen würden.
Das war jedenfalls bisher unsere Überlegung dabei. Wenn du es trotzdem gerne mit Übersetzung machen möchtest, dann sollte dein "Objekte gelöscht:" oben aber auch übersetzt werden.

Reply:
wird gemacht